src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
If you're referring to mine, well, yes, I wasn't really writing for long-term use...

As in NO_SUCCESS/FAILURE and MAYBE_SUCCESS/whatever, with UNKNOWN back to being only if "asked" about the wrong action? Sounds reasonable.

There are two categories of use cases that I can see:
-Supplied to the agent directly, for use as you suggest (and, more generally, an extension onto the idea of supplying an agent with what action it took last "turn");
-Supplied to the learning algorithm, to add more ability to distinguish A. actions that contributed to eventual success/failure, such as could be used by SARSA and similar (IIRC, the main HFO paper was using ball distance from goal for that?); and B. quality of overall success/failure (and the individual agent's maximum possible contributions to the former, as in proportion of actions that did not receive negative feedback). Another way to help with "B", which I am interested in taking a look at, is using the time to achieve a goal as a measure of how successful the offense was and how unsuccessful the defense was (another way to look at it is how much handicapping of the offense (by a lower time limit, as might happen at the end of a full match) it would take to change the outcome).